src/pages/ReportParticipantsPage.tsx

```diff
@@ -83,14 +83,28 @@ function ReportParticipantsPage({report, route}: ReportParticipantsPageProps) {
     const canSelectMultiple = isGroupChat && isCurrentUserAdmin && (isSmallScreenWidth ? selectionMode?.isEnabled : true);
     const [searchValue, setSearchValue] = useState('');
 
+    const chatParticipants = useMemo(() => getParticipantsList(report, personalDetails), [report, personalDetails]);
+
     useEffect(() => {
-        if (isFocused) {
+        if (selectedMembers.length === 0 || !canSelectMultiple) {
             return;
         }
-        setSelectedMembers([]);
-    }, [isFocused]);
 
-    const chatParticipants = getParticipantsList(report, personalDetails);
+        setSelectedMembers((prevSelectedMembers) => {
+            const updatedSelectedMembers = prevSelectedMembers.filter((accountID) => {
+                const isInParticipants = chatParticipants.includes(accountID);
+                const pendingChatMember = reportMetadata?.pendingChatMembers?.find((member) => member.accountID === accountID.toString());
+
+                const isPendingDelete = pendingChatMember?.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE;
+
+                // Keep the member only if they're still in the room and not pending removal
+                return isInParticipants && !isPendingDelete;
+            });
+
+            return updatedSelectedMembers;
+        });
+        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
```

Comment:
@daledah @s77rt Why do you need to skip these eslint rules? Could you please think of a way to avoid that? This then poses a risk of someone changing the useEffect logic and forgetting to add the correct dependencies. 

If its really not avoidable in your opinion, can you please add a clear explanation into a comment here why is that?

Reply:
The `useEffect` is complaining about the `selectedMembers` missing dependency, but we don't want to run the effect if `selectedMembers` changes. Technically we can avoid this problem if we create a `selectedMembersRef` where we can access the `selectedMembers` without having either in the dependency list and avoid the lint error however that's not worth it.

Now looking at the code again I can see that we can avoid the use of `selectedMembers` in the effect afterall as it's only used as an early-return exit.

cc @daledah